### PR TITLE
Add test and doc for ast-builder::statements::querying::data-aggregation

### PR DESCRIPTION
**Pull request by devgony**
## Goal 
Add test and doc for ast-builder::statements::querying::data-aggregation

### CODE REVIEW

File: docs/docs/ast-builder/statements/querying/data-aggregation.md [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+);
+```
+
+We will use this table for the subsequent examples.
+
+## Grouping and Counting
+
+To group records by a specific column and count the number of occurrences in each group, you can use the AST Builder's `group_by()` and `project()` methods.
+
+```sql
```
Comment (ever0de) [inline docs/docs/ast-builder/statements/querying/data-aggregation.md:23]:
```suggestion
```rust
```

File: docs/docs/ast-builder/statements/querying/data-aggregation.md [hunk 2]
```diff
@@ -1,5 +1,63 @@
...
+20  | 1
+30  | 2
+50  | 2
+```
+
+## Filtering Groups with HAVING
+
+You can further filter the groups based on specific conditions using the `having()` method. The `having()` method allows you to apply conditions to the grouped data.
+
+```sql
```
Comment (ever0de) [inline docs/docs/ast-builder/statements/querying/data-aggregation.md:45]: ditto

**Review by ever0de — APPROVED** [review]

**Review by panarch — APPROVED** [review]
Looks all nice! Thanks a lot 👍 👍

**Review by panarch — APPROVED** [review]
Looks all nice! Thanks a lot 👍 👍

**Pull request merged**